game_for_class.py

```diff
@@ -8,70 +8,46 @@
 # Start with the player's class
 class Character:
     """All the methods the game player can use"""
-    def __init__(self,
-                 name: str,
-                 hp: float,
-                 attack: float,
-                 defense: float,
-                 speed: float,
-                 team: int):
+    def __init__(self, name: str, hp: float, attack: float, defense: float, speed: float, team: int):
+
         self.name = name
         self.hp = hp
         self.attack = attack
         self.defense = defense
         self.speed = speed
         self.team = team
+        """Initializes the Character"""
 
     def is_alive(self) -> bool:
-        """Returns true if the character is alive"""
+
+        """returns true if the player is alive"""
         return self.hp > 0
 
-    def is_enemy(self, other_player) -> bool:
-        """Returns true if the player is an enemy"""
-        return other_player.team != self.team
+    def is_ally(self, other_player) -> bool:
+        """returns true if the other player is an ally"""
+        return other_player.team == self.team
 
-    def deal_damage(self, other_player: 'Character') -> 'Character':
-        """Deals damage to another Charater Object
+    def get_all_enemy_indices(self, players: List['Character']) -> List[int]:
```

Comment:
You may want to specify that you return the list of indices from a list of players where each index is an ally in the list of players.

Reply:
There's no allies in game_for_class it's just a 1 vs 1